src/preset_cli/cli/superset/sync/native/command.py

```diff
@@ -297,59 +310,115 @@ def native(  # pylint: disable=too-many-locals, too-many-arguments, too-many-bra
+    related_configs: Dict[str, Dict[Path, AssetConfig]] = {}
+
+    file_path = Path("progress.log")
+    if not file_path.exists():
+        file_path.touch()
+
+    with open(file_path, "r", encoding="utf-8") as log_file:
+        logs = yaml.load(log_file, Loader=yaml.SafeLoader) or {}
+
+    # Remove FAILED logs to re-try them
+    if "assets" in logs:
+        logs["assets"] = [
+            asset for asset in logs["assets"] if asset["status"] != "FAILED"
         ]
-        related_configs: Dict[str, Dict[Path, AssetConfig]] = {}
-        for resource_name, get_related_uuids in imports:
-            for path, config in configs.items():
-                if path.parts[1] != resource_name:
-                    continue
+    else:
+        logs["assets"] = []
+
+    assets_to_skip = {Path(log["path"]) for log in logs["assets"]} if logs else set()
+
+    for resource_name, get_related_uuids in imports:
+        for path, config in configs.items():
+            if path.parts[1] != resource_name or path in assets_to_skip:
+                continue
+
+            asset_configs = {path: config}
+            _logger.info("Importing %s", path.relative_to("bundle"))
 
-                asset_configs = {path: config}
+            try:
                 for uuid in get_related_uuids(config):
                     asset_configs.update(related_configs[uuid])
-
-                _logger.info("Importing %s", path.relative_to("bundle"))
                 contents = {str(k): yaml.dump(v) for k, v in asset_configs.items()}
-                if path not in imported:
-                    import_resources(contents, client, overwrite)
-                    imported.add(path)
-                    log.write(str(path) + "\n")
-                    log.flush()
+                import_resources(contents, client, overwrite)
+            except Exception:  # pylint: disable=broad-except
+                if not continue_on_error:
+                    write_logs_to_file(logs, file_path)
+                    raise
+
+                add_asset_to_log(path, config["uuid"], logs, assets_to_skip, "FAILED")
+                continue
+
+            add_asset_to_log(path, config["uuid"], logs, assets_to_skip, "SUCCESS")
+
+            related_configs[config["uuid"]] = asset_configs
+
+    if not continue_on_error or not any(
+        log["status"] == "FAILED" for log in logs["assets"]
+    ):
+        os.unlink(file_path)
+    else:
+        write_logs_to_file(logs, file_path)
 
-                related_configs[config["uuid"]] = asset_configs
 
-    os.unlink(checkpoint_path)
+def add_asset_to_log(
+    asset_path: Path,
+    asset_uuid: str,
+    logs: Dict[str, Any],
+    set_: Set[Path],
+    status: str,
+) -> None:
+    """
+    Adds a log entry to the logs to skip future occurrences.
+    """
+    log_entry = {
+        "path": str(asset_path),
+        "status": status,
+        "uuid": asset_uuid,
+    }
+    if "assets" not in logs:
+        logs["assets"] = []
```

Comment:
It's kind of old-school, but you can do this in 1 line:

```python
logs.setdefault("assets", [])
```

Reply:
hmm I'm actually thinking I might not need this as well. Going to remove it.